app/domain/services/prepare_clue_calculations/service.rb

```diff
@@ -498,63 +498,84 @@ def process
           end
         end
 
-        # Get uuids of records that will be updated
-        student_clue_calculation_values = student_clue_calculations.map do |scc|
-          [ scc.student_uuid, scc.book_container_uuid ]
+        if student_clue_calculations.empty?
+          updated_scc_uuids = []
+        else
+          # Get uuids of records that will be updated
+          student_clue_calculation_values = student_clue_calculations.map do |scc|
+            [ scc.student_uuid, scc.book_container_uuid ]
+          end
+          updated_scc_uuids = StudentClueCalculation.joins(
+            <<~JOIN_SQL
+              INNER JOIN (#{ValuesTable.new(student_clue_calculation_values)}) AS "values"
+                ("student_uuid", "book_container_uuid")
+                ON "student_clue_calculations"."student_uuid" = "values"."student_uuid"::uuid
+                  AND "student_clue_calculations"."book_container_uuid" =
+                    "values"."book_container_uuid"::uuid
+            JOIN_SQL
+          ).ordered.lock('FOR UPDATE').pluck(:uuid)
+
```

Comment:
does the `lock` do anything here?  I thought that for the lock to be effective you needed to use the same ActiveRecord object for the update later.

regardless, I don't see a later update but might be missing it.

Reply:
The import right below is the update (upsert).